spec/controllers/assignment_invitations_controller_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+RSpec.describe AssignmentInvitationsController, type: :controller do
+  describe 'GET #show' do
+    let(:invitation) { create(:assignment_invitation_with_assignment) }
+
+    describe 'unauthenticated request' do
+      it 'will redirect the new user to sign in with GitHub' do
+        get :show, id: invitation.key
+        expect(response).to redirect_to(login_path)
+      end
+    end
+
+    describe 'authenticated request' do
+      let(:user) { create(:user) }
+
+      before(:each) do
+        session[:user_id] = user.id
+      end
+
+      it 'will set the correct invitation' do
+        get :show, id: invitation.key
+        expect(assigns(:invitation)).to_not be_nil
+      end
+    end
+  end
+
+  describe 'GET #accept_invitation' do
+    let(:invitation)   { create(:assignment_invitation_with_assignment) }
+    let(:organization) { invitation.assignment.organization }
+
+    let(:user) { create(:user) }
+
+    before(:each) do
+      session[:user_id] = user.id
+
+      @stub_values = {
+        team_id:        12,
+        team_name:      'Team: 1',
+        user_login:     'user',
+        repo_id:        8_675_309,
+        repo_name:      "#{invitation.assignment.title}: 1",
+        full_repo_name: "user/#{invitation.assignment.title.parameterize}-1"
+      }
+    end
+
+    describe 'successful invitation' do
+      it 'will redeem the users invitation and return a successul json message' do
```

Comment:
Yeah it's just because there are a lot api calls that go along with this, I'm going to leave it for now and come back to the whole stubbing and testing at another date.